packages/@aws-cdk/core/test/fs/fs-fingerprint.test.ts

```diff
@@ -179,6 +178,46 @@ describe('fs fingerprint', () => {
     });
   });
 
+  describe('fingerprint cache', () => {
+    const largeString = ' '.repeat(16 * 1024 * 1024);
+    const largefile = path.join(__dirname, 'inode-fp.1');
+
+    // Create a very large test file
+    beforeAll(() => {
+      const file = fs.openSync(largefile, 'w');
+      // 512MB file
+      for (let i = 0; i < 32; i++) {
+        fs.writeSync(file, largeString);
+      }
+      fs.closeSync(file);
+    });
+
+    afterAll(() => {
+      fs.unlinkSync(largefile);
+    });
+
+    test('caches fingerprint results', () => {
+      const hash1 = FileSystem.fingerprint(largefile, {});
+      const startTime = Date.now();
+      const hash2 = FileSystem.fingerprint(largefile, {});
+      expect(Date.now() - startTime).toBeLessThan(500);
```

Comment:
Would be better here to assert that `fs.openSync()` has been called only once (then you don't have to write a test for a large file and rely on a delay)

```ts
const openSyncSpy = jest.spyOn(fs, 'openSync');

// ...

expect(openSpyncSpy).toHaveBeenCalledTimes(1);
```

Reply:
My relative unfamiliarity with JS test frameworks reveals itself 😅

I'll update the test to use this approach instead.